### PR TITLE
[#5061] Basic user and group CLI

### PR DESCRIPTION
**Pull request by justinmclean**
### What changes were proposed in this pull request?

Added basic user and group commands.

### Why are the changes needed?

For the CLI to support users and groups.

Fix: #5061

### Does this PR introduce _any_ user-facing change?

No, but extends CLI support.

### How was this patch tested?

Tested locally.

### CODE REVIEW

**Comment by justinmclean** [comment]
I am waiting on #5057 and #5059 to be merged, so those PRs changes are included.

**Comment by justinmclean** [comment]
@shaofengshi @jerryshao I've updated the PR with the main branch now that the second CLI PR has been merged - can you please review.

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoOptions.java [hunk 1]
```diff
@@ -64,12 +66,13 @@ public Options options() {
...
-            "g", PROVIDER, "provider one of hadoop, hive, mysql, postgres, iceberg, kafka"));
+            "t", PROVIDER, "provider one of hadoop, hive, mysql, postgres, iceberg, kafka"));
+    options.addOption(createArgOption("l", USER, "user name"));
+    options.addOption(createArgOption("g", GROUP, "group name"));
 
     // Properties option can have multiple values
     Option properties =
-        createArgOption("p", PROPERTIES, "comma separated property name/value pairs");
-    properties.hasArgs();
+        Option.builder("p").longOpt(PROPERTIES).desc("property name/value pairs").hasArgs().build();
```
Comment (shaofengshi) [inline clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoOptions.java:75]: "comma separated" is a necessary description for the properties I think, why not keep it?

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/DeleteGroup.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    this.group = group;
+  }
+
+  /** Delete a group. */
+  public void handle() {
+    boolean deleted = false;
+
+    try {
+      GravitinoClient client = buildClient(metalake);
+      deleted = client.removeUser(group);
```
Comment (shaofengshi) [inline clients/cli/src/main/java/org/apache/gravitino/cli/commands/DeleteGroup.java:52]: should be "removeGroup()" method?

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/CreateGroup.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+   * @param group The name of the group.
+   */
+  public CreateGroup(String url, boolean ignoreVersions, String metalake, String group) {
+    super(url, ignoreVersions);
+    this.metalake = metalake;
+    this.group = group;
+  }
+
+  /** Create a new group. */
+  public void handle() {
```
Comment (shaofengshi) [inline clients/cli/src/main/java/org/apache/gravitino/cli/commands/CreateGroup.java:46]: Just as previous PR, the "handle()" method should be marked as overwrites;

File: docs/cli.md [hunk 4]
```diff
@@ -321,3 +333,55 @@ gcli column list --metalake metalake_demo --name catalog_postgres.hr.departments
...
+```bash
+gcli user delete --metalake metalake_demo --user new_user
+```
+
+### Group commands
+
+#### Create a group
+
+```bash
+gcli group create --metalake metalake_demo --user new_group
```
Comment (shaofengshi) [inline docs/cli.md:368]: "--group"?

File: docs/cli.md [hunk 5]
```diff
@@ -321,3 +333,55 @@ gcli column list --metalake metalake_demo --name catalog_postgres.hr.departments
...
+```
+
+#### Display a group's details
+
+```bash
+gcli group details --metalake metalake_demo --user new_group
```
Comment (shaofengshi) [inline docs/cli.md:374]: same with the previous one

File: docs/cli.md [hunk 6]
```diff
@@ -321,3 +333,55 @@ gcli column list --metalake metalake_demo --name catalog_postgres.hr.departments
...
+
+```bash
+gcli group list --metalake metalake_demo
+```
+
+#### Delete a group
+
+```bash
+gcli group delete --metalake metalake_demo --user new_group
+```
```
Comment (shaofengshi) [inline docs/cli.md:387]: and this.

File: clients/cli/src/main/java/org/apache/gravitino/cli/Properties.java [hunk 7]
```diff
@@ -55,19 +55,21 @@ public Properties(String delimiter, String keyValueSeparator) {
    * <p>Each pair in the input string is split by the specified delimiter, and then each pair is
    * further split by the key-value separator.
    *
-   * @param input The input string containing name-value pairs.
+   * @param inputs An arrays of input strings containing name-value pairs.
    * @return A map of entries, where each entry represents a key-value pair from the input string.
    */
-  public Map<String, String> parse(String input) {
+  public Map<String, String> parse(String[] inputs) {
```
Comment (shaofengshi) [inline clients/cli/src/main/java/org/apache/gravitino/cli/Properties.java:61]: Is this change related with the "user and group" CLI? seems it has already been handled in previous PR, please double check whether this is a code merge issue.

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoOptions.java (hunk 1 above)
Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoOptions.java:75]: The option has been updated, and they no longer have to be comma separated - see the description in cli.md.

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/DeleteGroup.java (hunk 2 above)
Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/commands/DeleteGroup.java:52]: Fixed

File: docs/cli.md [hunk 8]
```diff
@@ -321,3 +333,55 @@ gcli column list --metalake metalake_demo --name catalog_postgres.hr.departments
...
+gcli user details --metalake metalake_demo --user new_user
+```
+
+#### List all users
+
+```bash
+gcli user list --metalake metalake_demo
+```
+
+#### Delete a users
```
[the remark this answers is not in the document]
Comment (justinmclean) [reply docs/cli.md:357]: "a user" is correct. While an is usually used before words that start with a vowel, there are lots of exceptions, including words that start slient letter e.g. honour or those whose first letter is not pronounced like a vowel e.g. user or useful (it more a "you" sound) or those where is it e.g. MRI machine (M is pronounced "em")

File: docs/cli.md (hunk 4 above)
Comment (justinmclean) [reply docs/cli.md:368]: Fixed

File: docs/cli.md (hunk 5 above)
Comment (justinmclean) [reply docs/cli.md:374]: Fixed

File: clients/cli/src/main/java/org/apache/gravitino/cli/Properties.java (hunk 7 above)
Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/Properties.java:61]: It not a code merge issue - the way properties have been handled have been improved

**Comment by justinmclean** [comment]
@shaofengshi all comments addressed this is ready for review again.

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoOptions.java (hunk 1 above)
Comment (shaofengshi) [reply clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoOptions.java:75]: I see; This change is irrelavant with current PR, now mixed with other changes; Next time please use a seperate PR, that will be helpful for review as well as code management (revert, cherry-pick, etc).

**Review by shaofengshi — APPROVED** [review]
LGTM
